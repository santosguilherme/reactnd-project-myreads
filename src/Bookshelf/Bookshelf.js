import React from 'react';
import PropTypes from 'prop-types';

import Book from '../Book/Book';

import './Bookshelf.css';


const Bookshelf = ({title, books, onUpdateBook}) => (
    <div className="bookshelf">
        <h2 className="bookshelf-title">
            {`${title} (${books.length})`}
        </h2>
        <div className="bookshelf-books">
            <ol className="books-grid">
                {books.map((book, index) => (
                    <li key={index}>
                        <Book
                            book={book}
                            onUpdateBook={onUpdateBook}
                        />
                    </li>
                ))}
            </ol>
        </div>
    </div>
);

Bookshelf.propTypes = {
    title: PropTypes.string.isRequired,
    books: PropTypes.arrayOf(
        PropTypes.shape({
            shelf: PropTypes.string.isRequired
        })
    ),
    onUpdateBook: PropTypes.func.isRequired
};

export default Bookshelf;