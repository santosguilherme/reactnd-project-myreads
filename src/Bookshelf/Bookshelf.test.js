import React from 'react';
import {shallow} from 'enzyme';

import Bookshelf from './Bookshelf';

it('renders without crashing', () => {
    const props = {
        title: 'Read',
        books: [
            {
                title: 'My Book',
                shelf: 'read',
                authors: ['Santos, Guilherme']
            }
        ],
        onUpdateBook: jest.fn()
    };
    const wrapper = shallow(
        <Bookshelf {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
});