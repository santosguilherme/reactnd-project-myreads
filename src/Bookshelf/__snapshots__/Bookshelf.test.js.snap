// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bookshelf
    books={
        Array [
            Object {
              "authors": Array [
                "Santos, Guilherme",
              ],
              "shelf": "read",
              "title": "My Book",
            },
          ]
    }
    onUpdateBook={[Function]}
    title="Read"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="bookshelf-title"
>
          Read (1)
</h2>,
        <div
          className="bookshelf-books"
>
          <ol
                    className="books-grid"
          >
                    <li>
                              <Book
                                        book={
                                                  Object {
                                                            "authors": Array [
                                                              "Santos, Guilherme",
                                                            ],
                                                            "shelf": "read",
                                                            "title": "My Book",
                                                          }
                                        }
                                        onUpdateBook={[Function]}
                              />
                    </li>
          </ol>
</div>,
      ],
      "className": "bookshelf",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Read (1)",
          "className": "bookshelf-title",
        },
        "ref": null,
        "rendered": "Read (1)",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <ol
            className="books-grid"
>
            <li>
                        <Book
                                    book={
                                                Object {
                                                            "authors": Array [
                                                              "Santos, Guilherme",
                                                            ],
                                                            "shelf": "read",
                                                            "title": "My Book",
                                                          }
                                    }
                                    onUpdateBook={[Function]}
                        />
            </li>
</ol>,
          "className": "bookshelf-books",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Book
                                book={
                                                Object {
                                                                "authors": Array [
                                                                  "Santos, Guilherme",
                                                                ],
                                                                "shelf": "read",
                                                                "title": "My Book",
                                                              }
                                }
                                onUpdateBook={[Function]}
                />
</li>,
            ],
            "className": "books-grid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <Book
                  book={
                                    Object {
                                                      "authors": Array [
                                                        "Santos, Guilherme",
                                                      ],
                                                      "shelf": "read",
                                                      "title": "My Book",
                                                    }
                  }
                  onUpdateBook={[Function]}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "book": Object {
                    "authors": Array [
                      "Santos, Guilherme",
                    ],
                    "shelf": "read",
                    "title": "My Book",
                  },
                  "onUpdateBook": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ol",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="bookshelf-title"
>
            Read (1)
</h2>,
          <div
            className="bookshelf-books"
>
            <ol
                        className="books-grid"
            >
                        <li>
                                    <Book
                                                book={
                                                            Object {
                                                                        "authors": Array [
                                                                          "Santos, Guilherme",
                                                                        ],
                                                                        "shelf": "read",
                                                                        "title": "My Book",
                                                                      }
                                                }
                                                onUpdateBook={[Function]}
                                    />
                        </li>
            </ol>
</div>,
        ],
        "className": "bookshelf",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Read (1)",
            "className": "bookshelf-title",
          },
          "ref": null,
          "rendered": "Read (1)",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <ol
              className="books-grid"
>
              <li>
                            <Book
                                          book={
                                                        Object {
                                                                      "authors": Array [
                                                                        "Santos, Guilherme",
                                                                      ],
                                                                      "shelf": "read",
                                                                      "title": "My Book",
                                                                    }
                                          }
                                          onUpdateBook={[Function]}
                            />
              </li>
</ol>,
            "className": "bookshelf-books",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Book
                                    book={
                                                      Object {
                                                                        "authors": Array [
                                                                          "Santos, Guilherme",
                                                                        ],
                                                                        "shelf": "read",
                                                                        "title": "My Book",
                                                                      }
                                    }
                                    onUpdateBook={[Function]}
                  />
</li>,
              ],
              "className": "books-grid",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <Book
                    book={
                                        Object {
                                                            "authors": Array [
                                                              "Santos, Guilherme",
                                                            ],
                                                            "shelf": "read",
                                                            "title": "My Book",
                                                          }
                    }
                    onUpdateBook={[Function]}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "book": Object {
                      "authors": Array [
                        "Santos, Guilherme",
                      ],
                      "shelf": "read",
                      "title": "My Book",
                    },
                    "onUpdateBook": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ol",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
